### PR TITLE
(SERVER-1206) Rename occurrences of puppet-server to puppetserver

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: dev-resources/logback-test.xml [hunk 1]
```diff
@@ -5,7 +5,8 @@
         </encoder>
     </appender>
 
-    <!--<logger name="log4j.logger.org.eclipse.jetty.server" level="warn"/>-->
+    <logger name="org.eclipse.jetty" level="warn"/>
+    <logger name="org.apache.http" level="warn"/>
```
Comment (justinstoller) [inline dev-resources/logback-test.xml]: Seems unrelated but innocuous... Highlighting so that other's can agree/disagree.

**Comment by justinstoller** [comment]
Here we're duplicating the contents of `puppet-server-lib` into `puppetserver-lib` rather than renaming it. Are there special considerations around backwards compatibility with that directory path? Is there a plan to remove the duplication?

**Comment by cprice404** [comment]
@justinstoller re: duplicate `puppet-server-lib`, no that sounds like some bad git-fu on my part.  Will fix.

**Comment by cprice404** [comment]
@justinstoller the duplicate puppet-server-lib dir should be fixed now.

File: src/ruby/puppetserver-lib/puppet/server/puppet_config.rb [hunk 2]
```diff
@@ -23,11 +23,7 @@ def self.initialize_puppet(puppet_config)
 
     # (SERVER-410) Cache features in puppetserver for performance.  Avoiding
     # the cache is intended for agents to reload features mid-catalog-run.
```
Comment (haus) [inline src/ruby/puppetserver-lib/puppet/server/puppet_config.rb]: @cprice404 This change and a few others make me think you have a stale master branch you're working from.

**Comment by cprice404** [comment]
@haus : rebased

File: dev-resources/logback-test.xml (hunk 1 above)
Comment (cprice404) [inline dev-resources/logback-test.xml]: Agree, unrelated, but it's been generally useful for me when debugging tests in many situations.  I made it when debugging these tests and forgot to remove it, but to me it seems worth leaving in.

**Comment by justinstoller** [comment]
Looks like the rebase needs rebasing :(

It might be best to do it once we've had a sign off or two, so you don't have to keep rebasing.

I set  `lein test :all` before I went to lunch on the merge of master and your branch (before you rebased or the commits were merged to master that are now requiring another rebase). I had two integration tests fail for me.

Rerunning just the tests that failed earlier on just your branch (not merging into master or dealing with the resultant conflict) they are still failing. I don't know what travis is going to tell us (does it not try to merge before running tests?) but these are the two tests that are failing locally for me:

lein test :only puppetlabs.general-puppet.general-puppet-int-test/test-complicated-external-command-execution

lein test :only puppetlabs.general-puppet.general-puppet-int-test/custom-trusted-oid-mapping-test

**Comment by camlow325** [comment]
Since @cprice404 is currently out, I'm going to close this PR out, rebase it on the latest from the master branch, and fix up any remaining tests that might still be failing.  I'll put up a replacement PR and link to this one.

**Comment by camlow325** [comment]
I submitted #1014 to replace this.

**Pull request closed without merging**